pylot/control/mpc/mpc_agent_operator.py

```diff
@@ -0,0 +1,333 @@
+            MPCAgentOperator.on_notification)
+        # Set no watermark on the output stream so that we do not
+        # close the watermark loop with the carla operator.
+        return [pylot.utils.create_control_stream()]
+
+    def on_waypoints_update(self, msg):
+        with self._lock:
+            self._waypoint_msgs.append(msg)
+
+    def on_can_bus_update(self, msg):
+        with self._lock:
+            self._can_bus_msgs.append(msg)
+
+    def on_pedestrians_update(self, msg):
+        with self._lock:
+            self._pedestrian_msgs.append(msg)
+
+    def on_vehicles_update(self, msg):
+        with self._lock:
+            self._vehicle_msgs.append(msg)
+
+    def on_traffic_lights_update(self, msg):
+        with self._lock:
+            self._traffic_light_msgs.append(msg)
+
+    def on_speed_limit_signs_update(self, msg):
+        with self._lock:
+            self._speed_limit_sign_msgs.append(msg)
+
+    def on_notification(self, msg):
+        # Get hero vehicle info.
+        can_bus_msg = self._can_bus_msgs.popleft()
+        vehicle_transform = can_bus_msg.data.transform
+        vehicle_speed = can_bus_msg.data.forward_speed
+
+        # Get waypoints.
+        waypoint_msg = self._waypoint_msgs.popleft()
+        wp_angle = waypoint_msg.wp_angle
+        wp_vector = waypoint_msg.wp_vector
+        wp_angle_speed = waypoint_msg.wp_angle_speed
+
+        waypoints = collections.deque(itertools.islice(waypoint_msg.waypoints, 0, 50))  # only take 50 meters
+
+        # Get ground pedestrian info.
+        pedestrians_msg = self._pedestrian_msgs.popleft()
+        pedestrians = pedestrians_msg.pedestrians
+
+        # Get ground vehicle info.
+        vehicles_msg = self._vehicle_msgs.popleft()
+        vehicles = vehicles_msg.vehicles
+
+        # Get ground traffic lights info.
+        traffic_lights_msg = self._traffic_light_msgs.popleft()
+        traffic_lights = traffic_lights_msg.traffic_lights
+
+        # Get ground traffic signs info.
+        speed_limit_signs_msg = self._speed_limit_sign_msgs.popleft()
+        speed_limit_signs = speed_limit_signs_msg.speed_signs
+        speed_factor, state = self.stop_for_agents(vehicle_transform.location,
+                                                   wp_angle,
+                                                   wp_vector,
+                                                   wp_angle_speed,
+                                                   vehicles,
+                                                   pedestrians,
+                                                   traffic_lights)
+
+        control_msg = self.get_control_message(waypoints, vehicle_transform, vehicle_speed, speed_factor, msg.timestamp)
+        self._logger.info("Throttle: {}".format(control_msg.throttle))
+        self._logger.info("Steer: {}".format(control_msg.steer))
+        self._logger.info("Brake: {}".format(control_msg.brake))
+        self._logger.info("State: {}".format(state))
```

Comment:
Can we change the verbosity to debug for these logs?

Reply:
Done